### PR TITLE
Add timeout to dev requirements and add kustomize to hack/tools

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Improve dev experience from feedback received from first time contributors:
- Add timeout to base requirements for development
- Add Kustomize to hack/tools and use that in Makefile
- Order tools in Makefile in alphabetical order

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Add timeout to dev requirements and add kustomize to hack/tools
```

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -51,8 +51,8 @@
    - [install instructions][kustomizelinux] on Linux
 6. Configure Python 2.7+ with [pyenv][pyenv] if your default is Python 3.x.
 7. Install make.
-
-[go]: https://golang.org/doc/install
+8. Install [timeout][timeout]
+   - `brew install coreutils && sudo ln -s /usr/local/bin/gtimeout /usr/local/bin/timeout` on MacOS.
```
Comment (CecileRobertMichon) [inline docs/development.md:55]: @ksubrmnn @sozercan do either of you know if we need equivalent instructions for windows wsl?

**Comment by CecileRobertMichon** [comment]
cc @jackfrancis

File: docs/development.md (hunk 1 above)
Comment (sozercan) [reply docs/development.md:55]: it's already in `/usr/bin/timeout` in WSL for me

Comment (sozercan) [reply docs/development.md:55]: if you meant the whole instructions, using Linux should be fine in WSL

**Comment by jackfrancis** [comment]
fwiw `brew install coreutils` got me a `timeout` command right off the bat. I didn't have to add a symlink to gtimeout

```
FrancisBookMS:aks-engine jackfrancis$ ls -la /usr/local/bin/timeout 
lrwxr-xr-x  1 jackfrancis  wheel  36 Apr 10 13:39 /usr/local/bin/timeout -> ../Cellar/coreutils/8.32/bin/timeout
FrancisBookMS:aks-engine jackfrancis$ ls -la /usr/local/bin/gtimeout 
lrwxr-xr-x  1 jackfrancis  wheel  37 Apr 10 13:39 /usr/local/bin/gtimeout -> ../Cellar/coreutils/8.32/bin/gtimeout
```

```
FrancisBookMS:aks-engine jackfrancis$ uname -a
Darwin MININT-ACNK2NL.northamerica.corp.microsoft.com 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
FrancisBookMS:aks-engine jackfrancis$ brew --version
Homebrew 2.2.12
Homebrew/homebrew-core (git revision d4eb6; last commit 2020-04-10)
```

In case that helps anyone...

**Comment by CecileRobertMichon** [comment]
> fwiw brew install coreutils got me a timeout command right off the bat. I didn't have to add a symlink to gtimeout

confirmed this works for me as well. Removed the symlink from the instructions.

**Comment by nader-ziada** [comment]
@cpanato are you waiting to verify something specific with the `hold`?

**Comment by cpanato** [comment]
@nader-ziada i'm not part of the `cluster-api-azure-reviewers` or `cluster-api-azure-maintainers` so i just add the hold to avoid mistakes :)

since you are a reviewer we can remove 😄 
/hold cancel

**Comment by nader-ziada** [comment]
@cpanato Thanks for all your awesome contributions to the project

**Pull request merged**